src/renderer/viz/expressions/Ramp.js

```diff
@@ -0,0 +1,265 @@
+import BaseExpression from './base';
+import { implicitCast, checkExpression, checkMaxArguments } from './utils';
+
+import RampImage from './RampImage';
+import RampGeneric from './RampGeneric';
+import { DEFAULT_OTHERS } from './constants';
+
+/**
+* Create a ramp: a mapping between an input (a numeric or categorical expression) and an output (a color palette or a numeric palette, to create bubble maps)
+*
+* Categories to colors
+* Categorical expressions can be used as the input for `ramp` in combination with color palettes. If the number of categories exceeds the number of available colors in the palette new colors will be generated by
+* using CieLAB interpolation.
+*
+* Categories to numeric
```

Comment:
Ok, now I see that my previous comment was precipitated, but I think we shouldn't try to explain case by case each combination because there are too many combinations (and this docs doesn't list every combination, as it forgets about images, classifiers, buckets and top)